modules/LongTaskAPI/src/main/java/org/gephi/utils/longtask/api/LongTaskExecutor.java

```diff
@@ -133,12 +133,12 @@ public LongTaskExecutor(boolean doInBackground) {
      *                               <code>taskName</code> is null
      * @throws IllegalStateException if a task is still executing at this time
      */
-    public synchronized void execute(LongTask task, final Runnable runnable, String taskName,
+    public synchronized Future<Object> execute(LongTask task, final Runnable runnable, String taskName,
```

Comment:
You shouldn't have to change the `LongTaskExectutor`, use the `execute(LongTask task, Callable<V> callable)` method instead

Reply:
Ah true ! Changed it to use the Callable version